cmd/controller/controller.go

```diff
@@ -133,8 +139,26 @@ func (c *Controller) setNodeName() error {
 	return nil
 }
 
+// WaitForDiskCRD will block till the CRDs are loaded
+// into Kubernetes
+func (c *Controller) WaitForDiskCRD() {
+	for {
+		_, err := c.ListDiskResource()
+		if err != nil {
+			glog.Errorf("Disk CRD is not available yet. Retrying after %v, error: %v", CRDRetryInterval, err)
+			time.Sleep(CRDRetryInterval)
+			continue
+		}
+		glog.Info("Disk CRD is available")
+		break
+	}
+}
+
 // Start is called when we execute cli command ndm start.
 func (c *Controller) Start() {
+	//Wait for Disk CRD to be loaded
+	c.WaitForDiskCRD()
```

Comment:
I think this is not the right place to wait for disk CRD as we are starting the probe (probe.start()) before the controller in the ndm start cli command. Can you verify the logs, you should be seeing "Disk CRD is available" after "disk CRD is pushed to etcd" logs?

Reply:
you are right @pawanpraka1 .. I will shift this to the instantiation of controller.